flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -762,14 +758,83 @@ private void handleDefaultNodeWithBlock(BlockStatementNode bodyNode) {
         startNode(FlowNode.Kind.EXPRESSION);
         Branch.Builder branchBuilder =
                 startBranch(Branch.BODY_LABEL, FlowNode.Kind.BODY, Branch.BranchKind.BLOCK, Branch.Repeatable.ONE);
-        for (StatementNode statement : bodyNode.statements()) {
-            statement.accept(this);
-            branchBuilder.node(buildNode());
-        }
+        analyzeBlock(bodyNode, branchBuilder);
         endBranch(branchBuilder, bodyNode);
         endNode(bodyNode);
     }
 
+    private void analyzeBlock(BlockStatementNode blockStatement, Branch.Builder thenBranchBuilder) {
+        for (StatementNode statement : blockStatement.statements()) {
+            genCommentNode(statement, thenBranchBuilder);
+            statement.accept(this);
+            thenBranchBuilder.node(buildNode());
+        }
+        genCommentNode(blockStatement.closeBraceToken(), thenBranchBuilder);
+    }
+
+    private void visitStatement(StatementNode statementNode) {
+        genCommentNode(statementNode);
+        statementNode.accept(this);
+    }
+
+    private LineRange getCommentPosition(Node commentAttachedNode) {
+        TextRange textRangeWithMinutiae = commentAttachedNode.textRangeWithMinutiae();
+        TextRange textRange = commentAttachedNode.textRange();
+        LinePosition startPos = textDocument.linePositionFrom(textRangeWithMinutiae.startOffset());
+        LinePosition endPos = textDocument.linePositionFrom(textRange.startOffset());
+
+        return LineRange.from(commentAttachedNode.lineRange().fileName(), startPos, endPos);
+    }
+
+    private String getComment(MinutiaeList minutiaes) {
+        int size = minutiaes.size();
+        List<String> commentLines = new ArrayList<>();
+        for (int i = 0; i < size; i++) {
+            String minutiae = minutiaes.get(i).text();
+            String[] splits = minutiae.split("// ");
+            if (splits.length >= 2) {
+                commentLines.add(splits[1]);
+            }
+        }
+        if (commentLines.isEmpty()) {
+            return "";
```

Comment:
There are scenarios where comments may be empty. For example:

```ballerina
public function emptyComment() {
    // 
    int i = 0;
}
```

In such cases, we should identify the comment as a node and set its value to `""`. However, our current approach overlooks these situations. The best solution is to return an optional string from this method and ignore the comment node when the method returns an empty optional value.